src/diffusers/pipelines/pipeline_utils.py

```diff
@@ -1240,8 +1333,95 @@ def load_module(name, value):
         # import it here to avoid circular import
         from diffusers import pipelines
 
+        final_device_map = None
+        if device_map is not None and device_map == "auto":
+            # Load each module in the pipeline on a meta device so that we can derive the device map.
+            init_empty_modules = {}
+            for name, (library_name, class_name) in init_dict.items():
+                if class_name.startswith("Flax"):
+                    raise ValueError("Flax pipelines are not supported for `device_map`.")
+
+                # Define all importable classes
+                is_pipeline_module = hasattr(pipelines, library_name)
+                importable_classes = ALL_IMPORTABLE_CLASSES
+                loaded_sub_model = None
+
+                # Use passed sub model or load class_name from library_name
+                if name in passed_class_obj:
+                    # if the model is in a pipeline module, then we load it from the pipeline
+                    # check that passed_class_obj has correct parent class
+                    maybe_raise_or_warn(
+                        library_name,
+                        library,
+                        class_name,
+                        importable_classes,
+                        passed_class_obj,
+                        name,
+                        is_pipeline_module,
+                    )
+                    with accelerate.init_empty_weights():
+                        loaded_sub_model = passed_class_obj[name]
+
+                else:
+                    loaded_sub_model = _load_empty_model(
+                        library_name=library_name,
+                        class_name=class_name,
+                        importable_classes=importable_classes,
+                        pipelines=pipelines,
+                        is_pipeline_module=is_pipeline_module,
+                        pipeline_class=pipeline_class,
+                        name=name,
+                        torch_dtype=torch_dtype,
+                        cached_folder=cached_folder,
+                        force_download=force_download,
+                        resume_download=resume_download,
+                        proxies=proxies,
+                        local_files_only=local_files_only,
+                        token=token,
+                        revision=revision,
+                    )
+
+                if loaded_sub_model is not None:
+                    init_empty_modules[name] = loaded_sub_model
+
+            # determine device map
+            # Obtain a sorted dictionary for mapping the model-level components
+            # to their sizes.
+            module_sizes = {
+                module_name: sum(compute_module_sizes(module, dtype=torch_dtype).values())
```

Comment:
We don't need to sum. You can get the total size by checking the value of the key `""`
```suggestion
                module_name: compute_module_sizes(module, dtype=torch_dtype)[""])
```